services/core-web/common/models/documents/document.ts

```diff
@@ -69,9 +79,26 @@ export class MineDocument {
     this.archived_by = jsonObject.archived_by;
     this.archived_date = jsonObject.archived_date;
     this.is_latest_version = jsonObject.is_latest_version ?? true;
+    this.category_code = this.determineCategoryCode(jsonObject);
     this.setCalculatedProperties(jsonObject);
   }
 
+  private determineCategoryCode(jsonObject: any): string | undefined {
```

Comment:
Little nitpick:
- to make this method available to extended classes, change from private to protected
- the project_summary_document_xref, etc, I think those are only for MajorMineApplicationDocuments, right?
- then we can use the appropriate type in the archived documents section, and it'll have the appropriate permissions based on the type.

Reply:
@taraepp Updated.